processor/samplingprocessor/probabilisticsamplerprocessor/testdata/config.yaml

```diff
@@ -2,8 +2,26 @@ receivers:
   examplereceiver:
 
 processors:
+  # The probabilistic_sampler sets trace sampling by hashing the trace id of
+  # each span and making the sampling decision based on the hashed value. It
+  # also implements the "sampling.priority" semantic convention as defined by
+  # OpenTracing. See
+  # https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table
+  # The "sampling.priority" semantics have priority over trace id hashing and
+  # can be used to control if given spans are sampled, ie.: forwarded, or not.
   probabilistic_sampler:
+    # the percentage rate at which traces are going to be sampled. Defaults to
+    # zero, i.e.: no sample. Values greater or equal 100 are treated as
+    # "sample all traces".
     sampling_percentage: 15.3
+    # hash_seed allows one to configure the hashing seed. This is important in
```

Comment:
When the sampling algo works on span inside the collector - will the `sampling.priority` be set on it for downstream to be able to read it?

Reply:
It can be configured separately on the [`attributesprocessor`](https://github.com/open-telemetry/opentelemetry-collector/blob/master/processor/attributesprocessor/testdata/config.yaml).

Reply:
How will the attirbuteProcessor know the seed value and the resulting hash value calculated from the TraceID?

Reply:
The calculated hash is not exposed (it is a predictable algo though). If desired the seed can be shared in the configuration as an environment variable so both processors can use the same value. I think that perhaps I'm missing some context

Reply:
`sampling.priority` is used to communicate calculated value of a hash. Typically from SDK to a dependent service (via tracestate-like propagation) and to collector for additional sampling and/or better visualization. If collector is running as a sidecar and calculates hash - it may be useful to communicate this information to collector which is actually collecting data. In case additional sampling needs to be implemented on the whole cluster for instance.

Reply:
I'm also very interested in how to propagate sampling.priority to all Spans in a trace. This code looks at each Span independently and so each Span would need this attribute applied potentially. What mechanism should be used to ensure this attribute flows to all Spans in a heterogenous environment if we don't use tracestate?

In our deployments, we trigger sampling at the hosts, although maybe we should. Everything is passed to the collector and sampled at 1% there. We do this so we can quickly adjust multi-cluster wide sampling to control our monitoring costs based on the ingestion volume. However, I want to be able communicate to the collector that a particular trace should always be sampled if some conditions were met; maybe we always want to sample certain kinds of monitoring traffic based on span name, etc.